tICA/scripts/feature_helpers/findOutliers.m

```diff
@@ -0,0 +1,6 @@
+function isOutliers = findOutliers(data)
+% must be 1d array
+madOutliers=findScaledMADOutliers(data);
+perOutliers=findFivePercentageOutliers(data);
+isOutliers=and(madOutliers, perOutliers);
```

Comment:
Why do two ways?  The reason matlab's percentile is different than the percentile implementation here is that matlab takes the additional step of using the percentile value as a threshold (and your test repeats `57` three times, which is why matlab 2022 doesn't exclude any of them).

Reply:
I thought this was what the `isoutlier(a,'percentile',[5,95])` does. So, to match with what it actually does, I should use the larger deviation (in this test case,  300-median from the 95% direction) to set the 5% boundary by `median-(300-median)`?

Reply:
Your test in 2022b, with the exact duplicate issue fixed:

```
>> isoutlier([56.99 59 60 100 59 58 57 58 300 61 62 60 62 58 57.01], 'percentile', [5, 95]) 

ans =

  1x15 logical array

   1   0   0   0   0   0   0   0   1   0   0   0   0   0   0
```

I don't think matlab was doing anything more complicated than "5th percentile is 57, remove anything below 57, but 57 is not below 57", and the exact duplicate values were what made it seem complicated.  We can do it that way to be less arbitrary about exact duplicates, but that also makes it less predictable about how many it will mark as outliers.

The question is what behavior we actually want, not what behavior matlab implements.  Let's not add complication to emulate what we guess matlab might be doing, when that may not be what we want.

Reply:
Then I think we should add a comment to explain the reasoning behind this complication, like "we want outlier detection in these spots to only detect things that fail the MAD-based test, but never detect more than 5% of the highest or lowest values".